### PR TITLE
Update to typedoc 0.8.0

### PR DESCRIPTION
**Pull request by EbXpJ6bp**
Since typedoc was outdated, I updated it.

I can not do complete debugging, but I confirmed that it worked on my clone site.

by typedoc 0.5.5
https://github.com/Microsoft/vscode-docs/blob/eb0f79a71ad79c463e898f30769167a572422af6/docs/extensionAPI/vscode-api.md

by typedoc 0.8.0
https://github.com/Microsoft/vscode-docs/blob/dea80ed88bc2801f3a7c158c0584f8b935fcda27/docs/extensionAPI/vscode-api.md

### CODE REVIEW

File: scripts/api-doc/doc.js [hunk 1]
```diff
@@ -347,13 +347,16 @@ function type2HtmlString(type) {
         case 'tuple':
             label = '[' + type.elements.map(type2HtmlString).join(', ') + ']';
             break;
+        case 'array':
+            label = type2HtmlString(type.elementType) + '[]';
+            break;
+        case 'unknown':
+            label = "<a class=\"type-unkwnon\">" + type.name + "</a>"; //https://github.com/TypeStrong/typedoc/pull/365
```
Comment (gregvanl) [inline scripts/api-doc/doc.js]: Is this a typo? type-unkwnon -> type-unknown?

File: scripts/api-doc/doc.ts [hunk 2]
```diff
@@ -382,15 +382,17 @@ function type2HtmlString(type: Type): string {
         case 'tuple':
             label = '[' + type.elements.map(type2HtmlString).join(', ') + ']';
             break;
+        case 'array':
+            label = type2HtmlString(type.elementType) + '[]'
+            break;
+        case 'unknown':
+            label = `<a class="type-unkwnon">${type.name}</a>`; //https://github.com/TypeStrong/typedoc/pull/365
```
Comment (gregvanl) [inline scripts/api-doc/doc.ts]: Actually I guess the typo is here in the TypeScript file

**Comment by EbXpJ6bp** [comment]
Thanks. Fix it.

**Comment by gregvanl** [comment]
Thanks!

**Pull request merged**